src/ast_def_macros.hpp

```diff
@@ -29,6 +29,11 @@ class LocalOption {
 };
 
 #define LOCAL_FLAG(name,opt) LocalOption<bool> flag_##name(name, opt)
+#define LOCAL_COUNT(name,opt) LocalOption<size_t> cnt_##name(name, opt)
+
+#define NESTING_GUARD(name) \
+  LocalOption<size_t> cnt_##name(name, name + 1); \
+  if (nestings > MAX_NESTING) throw Exception::NestingLimitError(pstate); \
```

Comment:
Should `nestings` be `name` here?

Reply:
No, the Macro is called as `NESTING_GUARD(nestings)`, therefore it will in turn insert a call to `LocalOption<size_t> cnt_nestings(nestings, nestings + 1)`. Meaning it will store the previous value and then increase it by one (reverting once local variable `cnt_nesting` goes out of scope).